### ISSUE TITLE
influxdn - Grafana stack is not scalable due to unsupported time roll up

### ISSUE THREAD

**Issue by PaulKuiper** [influxdb/influxdb#4605]
Influxdb has the nice feature to roll up time serie data into larger time buckets via continous queries. This can tremendously speed up a query that needs to display two years of data that was originally sampled at 1Hz. 

Grafana will however creates a select query on the original time serie and request a dynamic groupby time based on how far it is zoomed out. Influx will thus crunch all data points in that two year interval. There is no way in Grafana to specify to switch to the time rolled up continous query based on the groupby interval. This could potentially fixed in Grafana, however...

Grafana points out that Graphite automatically rollup its measurements as part of there measurement definition. An queries will thus automatically pick the best internally precalculated time roll up table to execute the query. Graphite therefore always answers within about 100ms. My experience with Influx is that it can take 30sec easily to return the result of a very zoomed out time serie. See https://github.com/grafana/grafana/issues/420 for more details. 

This is basically a very nasty interface problem between Grafana and Influxdb and I see no urgency on either side to solve it while it is the number one performance bottleneck in this popular stack as far as I'm concerned. 

The question:
Can the influx query engine use the knowledge its continous queries definitions to optimise the query engine path to use the pre calculated continous query data when a time groupby is requested on the original serie?

**Comment by beckettsean** [comment]
Highly related to https://github.com/influxdb/influxdb/issues/2625. Not quite a duplicate, but likely a fix for one addresses the other.

**Comment by beckettsean** [comment]
Tangentially related to https://github.com/influxdb/influxdb/issues/1994 as well.

**Comment by elnappo** [comment]
I also think this should be fixed in InfluxDB. Maybe we could use a dedicated tag which defines the function (mean, count, sum, ...) to use for aggregation? E.g.: 

    cpu,host=server01,region=uswest,aggregate=mean value=1 1434055562000000000

This would lead to better performance on all queries with include a GROUP BY statement, but increases the database size. It could be difficult to figure out what's the optimal size/interval for these precomputed roll ups. I don't know what's Graphite is doing here.

**Comment by sslupsky** [comment]
+1.  In my view, Grafana could address this by making a conditional query based on the time span.  Here are a couple marked up screen shots: 
![screen shot 2016-02-10 at 8 36 04 am](https://cloud.githubusercontent.com/assets/16888734/12952088/84735c2c-cfd2-11e5-9997-263f41ab63c2.png)
![screen shot 2016-02-10 at 8 36 04 am v2](https://cloud.githubusercontent.com/assets/16888734/12952094/86c4f8f0-cfd2-11e5-9896-b6a5fe8a93af.png)

In either case, the actual query could be modified to reference the proper continuous query for the given time span.

**Comment by sslupsky** [comment]
Here is an example of what I mean:

![influx continuous query mockup](https://cloud.githubusercontent.com/assets/16888734/12953355/58414326-cfd7-11e5-815d-c3dac8d92494.png)

**Comment by beckettsean** [comment]
Largely addressed by https://github.com/influxdata/influxdb/issues/6910

**Comment by friesoft** [comment]
Still relevant imho as intelligent rollups didn't land

**Comment by friesoft** [comment]
Still relevant as can be seen in #7198

**Comment by Apollon77** [comment]
Yes, I also would like to see a solution in Grafana because I do not think InfuxDB will add that feature soon :-(

**Comment by Apollon77** [comment]
I see that still relevant

**Comment by piwi447** [comment]
I like this simple proposal of having conditional query in Grafana but is it the right place to request this?
